source/common/grpc/google_grpc_utils.cc

```diff
@@ -0,0 +1,110 @@
+#include "common/grpc/google_grpc_utils.h"
+
+#include <atomic>
+#include <cstdint>
+#include <cstring>
+#include <string>
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/assert.h"
+#include "common/common/empty_string.h"
+#include "common/common/enum_to_int.h"
+#include "common/common/fmt.h"
+#include "common/common/macros.h"
+#include "common/common/stack_array.h"
+#include "common/common/utility.h"
+
+#include "absl/strings/match.h"
+
+namespace Envoy {
+namespace Grpc {
+
+struct BufferInstanceContainer {
+  BufferInstanceContainer(int ref_count, Buffer::InstancePtr buffer)
+      : ref_count_(ref_count), buffer_(std::move(buffer)) {}
+  std::atomic<uint32_t> ref_count_; // In case gPRC dereferences in a different threads.
+  Buffer::InstancePtr buffer_;
+
+  static void derefBufferInstanceContainer(void* container_ptr) {
+    auto container = reinterpret_cast<BufferInstanceContainer*>(container_ptr);
```

Comment:
`static_cast`?

Reply:
It seems that casting from or to void requires reinterpret_cast<>.

Reply:
I don't think so? pointer cast from or to `void*` should be `static_cast`. https://en.cppreference.com/w/cpp/language/static_cast, does build break?

Reply:
I could have sworn that it didn't compile before, but it is fine now.
Changed to static_cast<>().